models/iaf_tum_2000.h

```diff
@@ -0,0 +1,482 @@
+  }
+
+  // ----------------------------------------------------------------
+
+  /**
+   * @defgroup iaf_psc_exp_data
+   * Instances of private data structures for the different types
+   * of data pertaining to the model.
+   * @note The order of definitions is important for speed.
+   * @{
+   */
+  Parameters_ P_;
+  State_ S_;
+  Variables_ V_;
+  Buffers_ B_;
+  /** @} */
+
+  //! Mapping of recordables names to access functions
+  static RecordablesMap< iaf_tum_2000 > recordablesMap_;
+};
+
+
+inline size_t
+nest::iaf_tum_2000::send_test_event( Node& target, size_t receptor_type, synindex, bool )
+{
+  if ( target.get_model_id() != this->get_model_id() and target.is_off_grid() )
+  {
+    throw IllegalConnection( "iaf_tum_2000 neurons cannot be connected to precise spiking neurons." );
+  }
+
+  SpikeEvent e;
+  e.set_sender( *this );
+  return target.handles_test_event( e, receptor_type );
+}
+
+inline size_t
+iaf_tum_2000::handles_test_event( SpikeEvent& e, size_t receptor_type )
+{
+  if ( receptor_type > 1 )
+  {
+    throw UnknownReceptorType( receptor_type, get_name() );
+  }
+  else if ( receptor_type != 1 and e.get_sender().get_model_id() == this->get_model_id() )
+  {
+    throw IllegalConnection( "iaf_tum_2000 neurons must be connected via receptor_type 1." );
+  }
+  return receptor_type;
+}
+
+inline size_t
+iaf_tum_2000::handles_test_event( CurrentEvent&, size_t receptor_type )
+{
+  if ( receptor_type == 0 )
+  {
+    return 0;
+  }
+  else if ( receptor_type == 1 )
+  {
+    return 1;
+  }
+  else
+  {
+    throw UnknownReceptorType( receptor_type, get_name() );
+  }
+}
+
+inline size_t
+iaf_tum_2000::handles_test_event( DataLoggingRequest& dlr, size_t receptor_type )
+{
+  if ( receptor_type != 0 )
+  {
+    throw UnknownReceptorType( receptor_type, get_name() );
+  }
+  return B_.logger_.connect_logging_device( dlr, recordablesMap_ );
+}
+
+inline void
+iaf_tum_2000::get_status( DictionaryDatum& d ) const
+{
+  P_.get( d );
+  S_.get( d, P_ );
+  ArchivingNode::get_status( d );
+
+  ( *d )[ names::recordables ] = recordablesMap_.get_list();
+}
+
+inline void
+iaf_tum_2000::set_status( const DictionaryDatum& d )
+{
+  Parameters_ ptmp = P_;                       // temporary copy in case of errors
+  const double delta_EL = ptmp.set( d, this ); // throws if BadProperty
+  State_ stmp = S_;                            // temporary copy in case of errors
+  stmp.set( d, ptmp, delta_EL, this );         // throws if BadProperty
+
+  // We now know that (ptmp, stmp) are consistent. We do not
+  // write them back to (P_, S_) before we are also sure that
+  // the properties to be set in the parent class are internally
+  // consistent.
+  ArchivingNode::set_status( d );
+
+  // if we get here, temporaries contain consistent set of properties
+  P_ = ptmp;
+  S_ = stmp;
+}
+
+inline double
+iaf_tum_2000::phi_() const
+{
+  assert( P_.delta_ > 0. );
+  return P_.rho_ * std::exp( 1. / P_.delta_ * ( S_.V_m_ - P_.Theta_ ) );
+}
```

Comment:
Move to cpp file, it can be inlined there since it is only called from code in the cpp file.